pkgs/by-name/uc/ucc/package.nix

```diff
@@ -1,28 +1,51 @@
 {
-  stdenv,
-  lib,
-  fetchFromGitHub,
-  libtool,
-  automake,
   autoconf,
-  ucx,
+  automake,
   config,
-  enableCuda ? config.cudaSupport,
   cudaPackages,
+  fetchFromGitHub,
+  lib,
+  libtool,
+  stdenv,
+  ucx,
+  # Configuration options
   enableAvx ? stdenv.hostPlatform.avxSupport,
+  enableCuda ? config.cudaSupport,
   enableSse41 ? stdenv.hostPlatform.sse4_1Support,
   enableSse42 ? stdenv.hostPlatform.sse4_2Support,
-}:
+}@inputs:
+let
+  inherit (lib.attrsets) getLib;
+  inherit (lib.lists) optionals;
+  inherit (lib.strings) concatStringsSep optionalString;
+
+  inherit (cudaPackages)
+    cuda_cccl
+    cuda_cudart
+    cuda_nvcc
+    cuda_nvml_dev
+    cudaFlags
+    nccl
+    ;
+
+  stdenv = throw "Use effectiveStdenv instead";
+  effectiveStdenv = if enableCuda then cudaPackages.backendStdenv else inputs.stdenv;
+in
+effectiveStdenv.mkDerivation (finalAttrs: {
+  __structuredAttrs = true;
+  # TODO: The M4 file requires a unified CUDA installation and doesn't support splayed outputs:
+  # https://github.com/openucx/ucc/blob/0c0fc21559835044ab107199e334f7157d6a0d3d/config/m4/cuda.m4
+  # Enabling strictDeps will break the build when CUDA is enabled.
+  strictDeps = false;
```

Comment:
I don't quite understand how the former implies the latter?

Reply:
Good point! I need to reword it. 
Something is going on I don’t fully understand with `strictDeps`. When it is disabled, `configurePhase` succeeds and when it is enabled, `configurePhase` fails.
I believe it succeeds due to the non-separation of build/host dependencies, but need to find out whether that’s substantiated.

Reply:
So either some of the buildInputs or some of the nativeBuildInputs are missing. I guess we don't have to debug and fix this right away, but maybe make the phrasing more explicit